corehq/form_processor/backends/sql/processor.py

```diff
@@ -99,6 +100,29 @@ def hard_delete_case_and_forms(cls, domain, case, xforms):
         case.deleted = True
         publish_case_saved(case)
 
+    @classmethod
+    def new_form_from_old(cls, xml, xform, value_responses_map, user_id):
+        from corehq.form_processor.interfaces.processor import FormProcessorInterface
+        from corehq.form_processor.interfaces.dbaccessors import FormAccessors
+        interface = FormProcessorInterface(xform.domain)
+        existing_form = FormAccessors(xform.domain).get_with_attachments(xform.get_id)
+
+        new_xml = etree.tostring(xml)
+        form_json = convert_xform_to_json(new_xml)
+        new_form = interface.new_xform(form_json)
```

Comment:
can just be `cls.new_form(form_json)` since `interface` will just pass it through to that method anyway.